src/herder/FetchingEnvelopes.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+// Copyright 2018 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "herder/EnvelopeItemMap.h"
+#include "herder/NodesInQuorum.h"
+#include "xdr/Stellar-SCP.h"
+
+#include <map>
+#include <set>
+
+namespace stellar
+{
+
+/**
+ * Container for envelopes that are currently being fetched. Also keeps track of
+ * envelopes that are discarded and should be ignored. Only allows envelopes
+ * above certaing slot index that can be set by setMinimumSlotIndex.
+ */
+class FetchingEnvelopes
```

Comment:
I don't think `FetchingEnvelopes` is useful as an abstraction: it's 90% of what `PendingEnvelopes` is